Source/Core/DolphinQt/Config/Mapping/MappingButton.cpp

```diff
@@ -235,3 +235,9 @@ void MappingButton::mouseReleaseEvent(QMouseEvent* event)
     return;
   }
 }
+
+bool MappingButton::eventFilter(QObject* object, QEvent* event)
+{
+  const QEvent::Type event_type = event->type();
+  return event_type == QEvent::KeyPress || event_type == QEvent::KeyRelease;
```

Comment:
If this does work (untested) the button in the advanced mapping window needs the same treatment.